src/interaction-handlers/iam/autocomplete.ts

```diff
@@ -0,0 +1,42 @@
+import { InteractionHandler, InteractionHandlerTypes } from '@sapphire/framework';
+import type { AutocompleteInteraction } from 'discord.js';
+import { IAMCommand } from '../../commands/iam';
+
+export class AutocompleteHandler extends InteractionHandler {
+  public constructor(ctx: InteractionHandler.LoaderContext, options: InteractionHandler.Options) {
+    super(ctx, {
+      ...options,
+      interactionHandlerType: InteractionHandlerTypes.Autocomplete
+    });
+  }
+
+  public override async run(interaction: AutocompleteInteraction, result: InteractionHandler.ParseResult<this>) {
+    return interaction.respond(result);
+  }
+
+  public override async parse(interaction: AutocompleteInteraction) {
+    if (interaction.commandName !== 'iam') return this.none();
+    const fc = interaction.options.getFocused(true);
+    const roles = await interaction.guild.roles.fetch()
+    const rgx = /^[a-z]/g
+    const all_roles = Array.from(roles.values())
```

Comment:
you forgot to ensure that you don't return roles a user already has....

Reply:
Newest commit addresses this problem